core/src/main/clojure/xtdb/operator/scan.clj

```diff
@@ -430,7 +438,15 @@
                                (.columnTypes)
                                keys)])))
 
+    (allTableColNames [_ wm]
```

Comment:
this looks like it returns col-types rather than col-names?

if it were to return types, it's best to merge them with `types/merge-col-types` rather than just `merge`, but tbh I'd probably go with taking keys and `set/union`ing?